Makefile

```diff
@@ -296,7 +297,7 @@ $(ROM): $(ELF)
 $(ELF): $(O_FILES) $(LIBULTRA_LIB) $(LDSCRIPT) $(BUILD_DIR)/linker_scripts/$(VERSION)/hardware_regs.ld $(BUILD_DIR)/linker_scripts/$(VERSION)/undefined_syms.ld $(BUILD_DIR)/linker_scripts/$(VERSION)/pif_syms.ld
 	$(LD) $(LDFLAGS) -T $(LDSCRIPT) \
 		-T $(BUILD_DIR)/linker_scripts/$(VERSION)/hardware_regs.ld -T $(BUILD_DIR)/linker_scripts/$(VERSION)/undefined_syms.ld -T $(BUILD_DIR)/linker_scripts/$(VERSION)/pif_syms.ld \
-		-Map $(MAP) $(LIBULTRA_LIB) -o $@
+		-Map $(MAP) --whole-archive $(LIBULTRA_LIB) -o $@
```

Comment:
In theory we shouldn't need `--whole-archive` for libultra.
Which file we want linked but is not referenced?

Reply:
I was actually having trouble with this in AF, so wanted to do it here preemptively. However I do have a branch with a modified ultralib that matches all the o files, and if I don't use `--whole-archive` it has trouble with `us2dex_emu`, `parameters`, `mtxcatl, and `mtxcatf`.

Reply:
I haven't checked, but at the very least at least one symbol from `parameters` should always be used, so I don't understand why it isn't being properly linked.
Maybe the linker script generation is borken? Can you post it?